src/java.base/share/classes/java/lang/foreign/AddressLayout.java

```diff
@@ -105,6 +105,15 @@ sealed public interface AddressLayout extends ValueLayout permits ValueLayouts.O
     @CallerSensitive
     AddressLayout withTargetLayout(MemoryLayout layout);
 
+    /**
+     * Returns an address layout with the same carrier, alignment constraint, name and order as this address layout,
+     * but without any specified target layout. Thus, the returned address is a raw address.
+     *
+     * @return an address layout with same characteristics as this layout, but with no target layout.
+     * @see #targetLayout()
+     */
+    AddressLayout withoutTargetLayout();
```

Comment:
The term "raw address" is not defined elsewhere. I'd suggest to leave the last sentence out, and maybe replace it with something similar to what you did for "withoutName" (e.g. useful to compare, yada yada).

Reply:
And, no need to make this a restricted method, because this effectively turns a "more privileged" address layout into a "less privileged" one.